### ISSUE TITLE
Rule Request: [no-unsafe-ng-deep] Support flagging usage of `::ng-deep` in styles

### ISSUE THREAD

**Issue by NateRadebaugh** [angular-eslint/angular-eslint#2160]
**Description and reproduction of the issue**

> https://angular.dev/guide/components/styling#ng-deep
> Angular's emulated encapsulation mode supports a custom pseudo class, ::ng-deep. Applying this pseudo class to a CSS rule disables encapsulation for that rule, effectively turning it into a global style. **The Angular team strongly discourages new use of ::ng-deep.** These APIs remain exclusively for backwards compatibility.

I'd like to detect cases where `::ng-deep` is used in styles. In my case, I'd be OK allowing if inside of `:host`, but that could be considered as a configuration on this rule.

```JSON
{
  "rules": {
    "@angular-eslint/no-unsafe-ng-deep": "error"
  }
}
```

Disallowed:
```TS
@Component({
  standalone: true,
  selector: "app-xxx",
  templateUrl: "./xxx.component.html",
  styles: `
    ::ng-deep {
      button {
```

Allowed:
```TS
@Component({
  standalone: true,
  selector: "app-xxx",
  templateUrl: "./xxx.component.html",
  styles: `
    :host ::ng-deep {
      button {
```

**Versions**

| Package           |           Version|
|-----------------------------------|------------------|
| @angular-devkit/architect   | 0.1800.2 |
| @angular-devkit/core    |     18.0.2 |
| @angular-devkit/schematics   | 18.2.8 |
| @angular/build      |         18.2.12 |
| @angular/cdk        |         18.2.14 |
| @angular/cli         |        18.2.12 |
| @schematics/angular      |    18.2.12 |
| rxjs                   |      7.8.1 |
| typescript           |        5.4.5 |
| zone.js           |           0.14.10 |

**Comment by reduckted** [comment]
Where I work, we implemented something similar as a [Stylelint](https://stylelint.io/) rule. I think that (or whatever CSS linter you prefer) is the best place for it. Adding a rule to this package would only work for inline styles and would required CSS/SCSS parsing to be implemented.

**Comment by NateRadebaugh** [comment]
@reduckted interested in sharing a gist or repo?

Btw, Technically eslint does support CSS linting at this point

**Comment by reduckted** [comment]
Sorry, unfortunately I can't share it.

> Btw, Technically eslint does support CSS linting at this point

That's a good point. Perhaps when ESLint supports CSS -in-JavaScript (https://github.com/eslint/css/issues/30) 
It would be feasible to add a rule to this package.
